extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +184,120 @@ private void restartJobs() {
-                            String thingUid = getThing().getUID().toString();
-                            String typeId = getThing().getThingTypeUID().getId();
-                            JobDataMap jobDataMap = new JobDataMap();
-                            jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
-                            jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-daily");
-
-                            // dailyJob
-                            Trigger trigger = newTrigger().withIdentity("trigger-daily-" + typeId, thingUid)
-                                    .withSchedule(CronScheduleBuilder.cronSchedule("0 0 0 * * ?")).build();
-                            JobDetail jobDetail = newJob(getDailyJobClass())
-                                    .withIdentity("job-daily-" + typeId, thingUid).usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
-                            logger.info("Scheduled astro job-daily-{} at midnight for thing {}", typeId, thingUid);
-
-                            // startupJob
-                            trigger = newTrigger().withIdentity("trigger-daily-startup-" + typeId, thingUid).startNow()
-                                    .build();
-                            jobDetail = newJob(getDailyJobClass()).withIdentity("job-daily-startup-" + typeId, thingUid)
-                                    .usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
-
-                            if (isPositionalChannelLinked()) {
-                                // positional intervalJob
-                                jobDataMap = new JobDataMap();
-                                jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
-                                jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-positional");
-
-                                Date start = new Date(System.currentTimeMillis() + (thingConfig.getInterval()) * 1000);
-                                trigger = newTrigger().withIdentity("trigger-positional", thingUid).startAt(start)
-                                        .withSchedule(simpleSchedule().repeatForever()
-                                                .withIntervalInSeconds(thingConfig.getInterval()))
-                                        .build();
-                                jobDetail = newJob(PositionalJob.class).withIdentity("job-positional", thingUid)
-                                        .usingJobData(jobDataMap).build();
-                                quartzScheduler.scheduleJob(jobDetail, trigger);
-                                logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
-                                        thingConfig.getInterval(), thingUid);
-                            }
+                    // Startup Job
+                    Thread thread = new Thread(dailyJob);
```

Comment:
As soon as the jobs are finished, the threads will be eligible to be garbage collected. It will happen asynchronously. I can call it synchronously by calling `job.run() ` but it could block if in future we change the execution to contain some time consuming operations. That's why I preferred to use a different thread. But if you feel that it is not needed, I can change it too synchronous execution.